ios/Classes/Embeds/AppcuesFrameViewFactory.swift

```diff
@@ -0,0 +1,195 @@
+    }
+
+    func create(
+        withFrame frame: CGRect,
+        viewIdentifier viewId: Int64,
+        arguments args: Any?
+    ) -> FlutterPlatformView {
+        return AppcuesPlatformView(
+            plugin: plugin,
+            frame: frame,
+            viewIdentifier: viewId,
+            arguments: args,
+            binaryMessenger: messenger)
+    }
+
+    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
+    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
+          return FlutterStandardMessageCodec.sharedInstance()
+    }
+}
+
+class AppcuesPlatformView: NSObject, FlutterPlatformView {
+    private var _view: UIView
+
+    init(
+        plugin: SwiftAppcuesFlutterPlugin,
+        frame: CGRect,
+        viewIdentifier viewId: Int64,
+        arguments args: Any?,
+        binaryMessenger messenger: FlutterBinaryMessenger?
+    ) {
+        let args = args as? [String: Any]
+
+        _view = WrapperView(
+            plugin: plugin,
+            binaryMessenger: messenger,
+            frameID: args?["frameId"] as? String,
+            viewIdentifier: viewId
+        )
+        super.init()
+    }
+
+    func view() -> UIView {
+        return _view
+    }
+}
+
+
+class WrapperView: UIView, FlutterStreamHandler {
+
+
+    private let plugin: SwiftAppcuesFlutterPlugin
+    private var eventChannel: FlutterEventChannel?
+    private var eventSink: FlutterEventSink?
+
+    var frameID: String? = nil
+
+    private weak var frameViewController: AppcuesFrameVC?
+
+    init(
+        plugin: SwiftAppcuesFlutterPlugin,
+        binaryMessenger messenger: FlutterBinaryMessenger?,
+        frameID: String?,
+        viewIdentifier viewId: Int64
+    ) {
+        self.plugin = plugin
+        self.frameID = frameID
+
+        if let messenger = messenger {
+            eventChannel = FlutterEventChannel(name: "com.appcues.flutter/frame/\(viewId)", binaryMessenger: messenger)
+        }
+
+        super.init(frame: .zero)
+
+        eventChannel?.setStreamHandler(self)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if let frameViewController = frameViewController {
+            frameViewController.view.frame = bounds
+        } else {
+            setupFrame()
+        }
+    }
+
+    private func setupFrame() {
+        guard let parentVC = parentViewController,
+              let frameID = frameID else {
+            return
+        }
+
+        let frameVC = AppcuesFrameVC(parentView: self)
+        parentVC.addChild(frameVC)
+        addSubview(frameVC.view)
+        frameVC.view.frame = bounds
+        frameVC.didMove(toParent: parentVC)
+        self.frameViewController = frameVC
+
+        plugin.register(frameID: frameID, for: frameVC.frameView, on: frameVC)
```

Comment:
this is the call that ends up going back to the `Appcues` implementation to register the frame - the Appcues implementation is held by the plugin, and must be initialized by the time the frame is used, or this will not get registered.